packages/components/src/modal/stories/index.tsx

```diff
@@ -1,33 +1,57 @@
 /**
  * External dependencies
  */
-import { boolean, text } from '@storybook/addon-knobs';
+import type { ComponentStory, ComponentMeta } from '@storybook/react';
 
 /**
- * Internal dependencies
+ * WordPress dependencies
  */
-import Button from '../../button';
-import Icon from '../../icon';
-import Modal from '../';
+import { useState } from '@wordpress/element';
 
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
-import { useState } from '@wordpress/element';
-import { wordpress } from '@wordpress/icons';
+import Button from '../../button';
+import Modal from '../';
+import type { ModalProps } from '../types';
 
-export default {
-	title: 'Components/Modal',
+const meta: ComponentMeta< typeof Modal > = {
 	component: Modal,
+	title: 'Components/Modal',
+	argTypes: {
+		children: {
+			control: { type: null },
+		},
+		onKeyDown: {
+			control: { type: null },
+		},
+		focusOnMount: {
+			control: { type: 'boolean' },
+		},
+		role: {
+			control: { type: 'text' },
+		},
+		onRequestClose: {
+			action: 'onRequestClose',
+		},
+	},
 	parameters: {
-		knobs: { disable: false },
+		controls: { expanded: true },
+		docs: { source: { state: 'open' } },
```

Comment:
I guess we can keep the source collapsed for this one because it isn't very helpful?

Reply:
Nice spot! Also made it so the canvas now reads `No code available`, since the code just showed a button didn't make sense.